src/CleanRegex/Internal/PatternStrings.php

```diff
@@ -46,6 +52,7 @@ private function predefinition($pattern): Predefinition
         if ($pattern instanceof Pattern) {
             return new IdentityPredefinition(new Definition($pattern->delimited()));
         }
+        // @phpstan-ignore-next-line Don't trust library user's input
         throw InvalidArgument::typeGiven("PatternList can only compose type Pattern or string", new ValueType($pattern));
```

Comment:
I definitely can't stand `// @phpstan-ignore-line Don't trust library user's input` comments after each `InvalidArgumentException`. 

So we need to disable it somehow, or ignore it globally. 

This is part of the reasons why I didn't add phpstan to the project yet. PhpStan holds some incorrect assumptions, like that fact that throwing `InvalidArgumentException` isn't desired, but that's the part of library interface. I dislike the idea that it's *assumed* it's incorrect and we must ignore it. The whole assumption is incorrect, since exceptions in T-Regx library are expected.

Reply:
It may be that PHPStan is designed partly to be a fill-in for a compiler, to simulate properly-typed code ("I know all type bugs before I start the code"). T-Regx has to support the PHP ways.